file-formats/dobj/type/zif_aff_dobj_v1.intf.abap

```diff
@@ -0,0 +1,61 @@
+INTERFACE zif_aff_dobj_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Structure Definition</p>
+    "! Structure Definition
+    BEGIN OF ty_structure_definition,
+      "! <p class="shorttext">Structure Definition</p>
+      "! Parent Table
+      "! $required
+      record_no       TYPE n LENGTH 4,
+      "! <p class="shorttext">Parent Table</p>
+      "! Parent Table
+      "! $required
+      parent_table    TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Dependent table</p>
+      "! Dependent table
+      "! $required
+      dependent_table TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_structure_definition,
+    "! <p class="shorttext">Structure Definitions</p>
+    "! Structure Definitions
+    ty_structure_definitions TYPE SORTED TABLE OF  ty_structure_definition WITH UNIQUE KEY record_no.
+
+
+  TYPES:
+    "! <p class="shorttext">Data destruction object</p>
+    "! $required
+    BEGIN OF ty_dobj_details,
+      "! <p class="shorttext">Destruction Program</p>
+      "! Destruction Program
+      "! $required
+      destr_program        TYPE zif_aff_types_v1=>ty_description_60,
```

Comment:
Is the type "description" intended?

Reply:
Have tried using the data types exposed by the interface if_aff_types_v1. The length of this more than the length of the actual data element used in ILM framework. Nevertheless, have used it as it can contain the description of the data destruction objects

Reply:
I am not sure. Shall the field contain a description or the name of the destruction program?

If it should contain the description, should we add the term "description" to the field?
If it should contain the object's name, we should use a different type with the exact length of your data element (i.e. by using a self-defined type or `c LENGTH <your field length>`. Does this make sense?